src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/StepsToCompleteController.java

```diff
@@ -42,30 +44,54 @@ protected String getTemplateName() {
 
     @GetMapping
     public String getStepsToComplete(@PathVariable String companyNumber,
-            Model model) {
-
-        addBackPageAttributeToModel(model, companyNumber);
+                                     @PathVariable Optional<String> transactionId,
+                                     @PathVariable Optional<String> companyAccountsId,
+                                     Model model) {
+
+        if (transactionId.isPresent() && companyAccountsId.isPresent()) {
+            addBackPageAttributeToModel(model, companyNumber, transactionId.get(), companyAccountsId.get());
+        } else {
+            addBackPageAttributeToModel(model, companyNumber);
+        }
 
         return getTemplateName();
     }
 
     @PostMapping
     public String postStepsToComplete(@PathVariable String companyNumber,
-            HttpServletRequest request) {
-
-        try {
-            String transactionId = transactionService.createTransaction(companyNumber);
+                                      @PathVariable Optional<String> transactionId,
+                                      @PathVariable Optional<String> companyAccountsId,
+                                      HttpServletRequest request) {
+
+        boolean isExistingTransaction = false;
+        String transactionID = "";
+        String companyAccountsID = "";
+
+        if (transactionId.isPresent() && companyAccountsId.isPresent()) {
+            transactionID = transactionId.get();
+            companyAccountsID = companyAccountsId.get();
+            isExistingTransaction = true;
+        }
 
-            String companyAccountsId = companyAccountsService.createCompanyAccounts(transactionId);
+        if (!isExistingTransaction) {
+            try {
+                transactionID = transactionService.createTransaction(companyNumber);
+                companyAccountsID = companyAccountsService.createCompanyAccounts(transactionID);
 
-            smallFullService.createSmallFullAccounts(transactionId, companyAccountsId);
+            } catch (ServiceException e) {
 
-            statementsService.createBalanceSheetStatementsResource(transactionId, companyAccountsId);
+                LOGGER.errorRequest(request, e.getMessage(), e);
+                return ERROR_VIEW;
+            }
+        }
 
-            transactionService.createResumeLink(companyNumber, transactionId, companyAccountsId);
+        try {
```

Comment:
You can wrap the whole thing in a single try / catch; the catch behaves the same in both blocks